### ISSUE TITLE
WC_AJAX method remove_variation is not defined

### ISSUE THREAD

**Issue by etoler1841** [woothemes/woocommerce#30482]
Please provide us with the information requested in this bug report. Without these details, we won't be able to fully evaluate this issue. 
Bug reports lacking detail, or for any other reason than to report a bug, may be closed without action.

**Prerequisites (mark completed items with an [x]):**
- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
In the class definition file for `WC_AJAX`, the AJAX event `remove_variation` is defined. This results in an action being added to the `wp_ajax_woocommerce_remove_variation` hook, which should execute the `remove_variation()` method of the `WC_AJAX` class. However, this method is not defined.

**Expected behavior**
Either the `WC_AJAX::remove_variation()` method should be defined or the `remove_variation` handler should be removed.

**Actual behavior**
The method is not defined in the class. I am not sure if the matching action is ever executed, as I discovered this inconsistency while browsing the code. 

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
No execution error has been discovered yet, but the bug exists at `includes/class-wc-ajax.php:128`.

**Screenshots**

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use the [WooCommerce System Status Report](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/) to help us evaluate the issue. 
Without this report we won't be able to fully evaluate this issue.
<details>
```
### WordPress Environment ###

WordPress address (URL): http://bam.local
Site address (URL): http://bam.local
WC Version: 5.4.2
REST API Version: ✔ 5.4.2
WC Blocks Version: ✔ 5.1.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.3.1
Log Directory Writable: ✔
WP Version: 5.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.29 (Ubuntu)
PHP Version: 7.4.20
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.58.0
OpenSSL/1.1.1

SUHOSIN Installed: –
MySQL Version: 5.7.34-0ubuntu0.18.04.1
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.5.2
WC Database Prefix: [REMOVED]
Total Database Size: 710.39MB
Database Data Size: 569.03MB
Database Index Size: 141.36MB
[REMOVED]

### Post Type Counts ###

action_monitor: 1269
attachment: 41488
custom_css: 7
et_pb_layout: 15
is_search_form: 1
nav_menu_item: 93
nf_sub: 67
oembed_cache: 34
page: 91
post: 26
product: 8694
product_variation: 1025
revision: 2180
shop_coupon: 5
shop_order: 280
shop_order_refund: 31
testimonial: 13
woocustomemails: 1
wp_block: 3
wp-rest-api-log: 2205
wpforms: 1

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (18) ###

BAM Testimonials: by Eric Toler  – 1.1.2
Classic Editor: by WordPress Contributors – 1.6
JWT Auth: by Useful Team – 1.4.2
Login Logo: by Mark Jaquith – 0.10.0
Podium: by Podium – 1.0.0
Redis Object Cache: by Till Krüss – 2.0.20
SwingDance: by Eric Toler  – 1.0.0
SwingDance Category Mapping: by Eric Toler, Anthony Rickard – 0.0.3
Woo Custom Emails Per Product: by Alex Mustin – 2.2.9
Variation Images Gallery for WooCommerce: by RadiusTheme – 2.0.1.1
WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.1.2
WooCommerce Shipping & Tax: by WooCommerce – 1.25.16 (update to version 1.25.17 is available)
WooCommerce - ShipStation Integration: by WooCommerce – 4.1.42 (update to version 4.1.43 is available)
WooCommerce: by Automattic – 5.4.2 (update to version 5.5.2 is available)
Yoast SEO: by Team Yoast – 16.6.1
WP Mail SMTP: by WPForms – 2.9.0
YITH WooCommerce Ajax Product Filter: by YITH – 4.1.0

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #11580 - /shop/
Cart: #2824 - /cart/
Checkout: #2825 - /checkout/
My account: #2826 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Blues Angel Music
Version: 1.0.10d
Author URL: https://github.com/bluesangelmusic
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: /home/eric/dev/wc-swingdance/wp-content/plugins/woo-product-variation-gallery/templates/product-images.php
/home/eric/dev/wc-swingdance/wp-content/plugins/woo-product-variation-gallery/templates/product-thumbnails.php


### Action Scheduler ###

Complete: 2,628
Oldest: 2021-07-19 10:49:17 -0500
Newest: 2021-08-11 14:34:19 -0500

Failed: 113,365
Oldest: 2020-05-02 17:08:19 -0500
Newest: 2021-06-19 00:14:57 -0500

Pending: 4
Oldest: 2021-08-11 14:34:23 -0500
Newest: 2021-08-17 16:34:07 -0500


### Status report information ###

Generated at: 2021-08-11 14:35:05 -05:00
```
</details>

**Comment by juliaamosova** [comment]
Hi @etoler1841,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by barryhughes** [comment]
Thanks for flagging!

Seems like this may be a remnant of [this change](https://github.com/woocommerce/woocommerce/commit/2c1c9896c5e5cdc8223c2ef253c188520b3e074c) and we simply forgot to clean this part up. Removing it—that is, removing the `remove_variation` (singular) entry from the list of ajax events—should be safe to do.

**Comment by san-coding** [comment]
Hello, I am new to open source , could you please assign this to me

**Comment by barryhughes** [comment]
Hi @san-coding—please feel free to submit a PR for this (and mention the issue number in the PR description). No need to worry about assignment :-)
